ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -411,7 +411,12 @@ private ByteBuffer getStoreKeyBytes(StoreKey key) {
    */
   private void generateBloomFilterAndPersist() throws StoreException {
     int numOfIndexEntries = numberOfEntries(serEntries);
-    bloomFilter = FilterFactory.getFilter(numOfIndexEntries, config.storeIndexBloomMaxFalsePositiveProbability);
+    // The number of elements that used to create bloom filter should be no less than "storeIndexMaxNumberOfInmemElements"
```

Comment:
rephrase this to say that this is a workaround since we found higher than intended false positive rates with small bloom filter sizes instead of saying it "should be no less than ...".